### PR TITLE
[4.0] npm cli tools updates

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Update some critical parts (eslint)
- moved the webcomponents folder to `build/media/webcomponents`
- copied the folders: media/vendor/tinyMCE/langs, media/vendor/tinyMCE/templates and media/vendor/jquery-ui to build/media/vendor/...  The first 2 folders are actually code that belongs to the Joomla project. The latter is just because jquery-ui is way to outdated to even care to fetch it properly (has been marked for removal, anyways).

- replaced the chalk with kleur (a highlighting script for the command line)
- removed the --compilecescss and the relative file.
- Refactor the compilecejs to be capable to produce inline css and also the relative files (the first one is required for the webcomponents, the latter for the custom elements)

Commands:
```js
/* *
 * To get the complete functional media folder please run
 *
 * npm install
 *
 * For dedicated tasks, please run:
 * node build.js --installer      === will create the error page (for unsupported PHP version)
 * node build.js --update         === will clean the media/vendor folder and then fetch the dependencies from source
 * node build.js --compilejs      === will transpile ES6 files and also uglify the ES6,ES5 files
 * node build.js --compilecejs    === will compile all the given CE or WC with their relative css files
 * node build.js --compilecss   === will compile all the scss defined files and also create a minified version of the css
 *
 */
```

### Testing Instructions

just try the commands

### Documentation Changes Required

### CODE REVIEW

**Comment by wilsonge** [comment]
```
Prefixing for:  [ 'last 2 versions', 'not ie < 11' ]
Without `from` option PostCSS could generate wrong source map and will not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning.
(node:2251) UnhandledPromiseRejectionWarning: ReferenceError: chalk is not defined
    at cleaner.process.then.then.then (/Users/george/Sites/joomla-cms/build/build-modules-js/compilescss.js:95:25)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:2251) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2251) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by dgrammatiko** [comment]
about: 
`Without `from` option PostCSS could generate wrong source map and will not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning.`

That is a warning that map needs a source file, but we don't use maps (at least for CE's)

The other was a stupid mistake on search-replace.

File: build/build-modules-js/settings.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "settings": {
     "browsers": [
-      "last 1 version"
+      "last 2 versions",
```
Comment (C-Lodder) [inline build/build-modules-js/settings.json]: I believe this will make autoprefixer add bulk to the css-grid CSS as it will need to support Edge 16.....dont know why you'd want to support the last 2 versions anyway

Comment (dgrammatiko) [reply build/build-modules-js/settings.json]: checking the produced css I didn't see anything more added, but this is just a flag, we can switch any time it if that is needed.
To defend the "why" part of your question:
Although the evergreen browsers (ie auto-updated) are supposed to always be on the latest version I can reassure you that this is an intention the reality is a little bit different. Anyways the good part with browserlist is that we can tweak it on our requirements before each release, it allows us to be dynamic. Maybe that is a good spot that the production team should investigate how the supported browsers will be adjusted for the lifespan of J4. 

TL;DR I will cange the flag back to 1 although this little discussion will be good to be the sparting point of a conversation in the production team to define a realistic/viable plan for the css/js support of the J4.

**Comment by wilsonge** [comment]
So why are we moving from chalk to kleur? According to kleur's readme

```
Aside from the performance boost, `kleur` exists as a separate module because I've removed some of `ansi-colors`'s defining features, like bright color variants and symbols. It's tailor-made for my needs and experimentation.
```

I'm not really seeing why a module that recommends somebody else's module because it's for experimentation is something we'd want to use. I don't see the need for a small performance boost for cli highlighting in a dev tool where css and js compilation is going to take magnitudes longer than cli

**Comment by dgrammatiko** [comment]
Actually, I'm gonna ditch the whole coloured output from the cli, there's no reason to get more dependencies for such things

**Comment by dgrammatiko** [comment]
@wilsonge highlighting completely removed
